core/src/main/java/org/jdbi/v3/core/mapper/ColumnMappers.java

```diff
@@ -184,23 +184,22 @@ public ColumnMappers register(QualifiedColumnMapperFactory factory) {
      * @return a ColumnMapper for the given type, or empty if no column mapper is registered for the given type.
      */
     @Beta
+    @SuppressWarnings({ "unchecked", "rawtypes" })
     public <T> Optional<ColumnMapper<T>> findFor(QualifiedType<T> type) {
         // ConcurrentHashMap can enter an infinite loop on nested computeIfAbsent calls.
         // Since column mappers can decorate other column mappers, we have to populate the cache the old fashioned way.
         // See https://bugs.openjdk.java.net/browse/JDK-8062841, https://bugs.openjdk.java.net/browse/JDK-8142175
-        @SuppressWarnings("unchecked")
-        ColumnMapper<T> cached = (ColumnMapper<T>) cache.get(type);
+        Optional<ColumnMapper<T>> cached = (Optional) cache.get(type);
 
         if (cached != null) {
-            return Optional.of(cached);
+            return cached;
         }
 
-        Optional<ColumnMapper<T>> mapper = factories.stream()
+        Optional<ColumnMapper<T>> mapper = (Optional) factories.stream()
```

Comment:
I'd rename this to `mapperHit` or something, since it's not just an instance of a mapper anymore

Reply:
This isn't actually changing the type - this was always how it was.  `mapperHit` is a little bit misleading because this is *not* a hit case -- it's actually the "miss" case where we look it up in the factories.

Reply:
Myeah I guess I was using `hit` a bit dubiously in my mind, having it also mean "miss" :p I figured you'd come up with a better word that means "hit or miss"